pkg/cmd/cli/cmd/buildlogs.go

```diff
@@ -59,39 +54,9 @@ func NewCmdBuildLogs(fullName string, f *clientcmd.Factory, out io.Writer) *cobr
 }
 
 // RunBuildLogs contains all the necessary functionality for the OpenShift cli build-logs command
-func RunBuildLogs(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, opts api.BuildLogOptions, args []string) error {
+func RunBuildLogs(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, opts api.BuildLogOptions, args []string) error {
 	if len(args) != 1 {
-		// maximum time to wait for a list of builds
-		timeout := 800 * time.Millisecond
-		// maximum number of builds to list
-		maxBuildListLen := 10
-		ch := make(chan error)
-		go func() {
-			// TODO fetch via API no more than maxBuildListLen builds
-			builds, err := getBuilds(f)
-			if err != nil {
-				return
-			}
-			if len(builds) == 0 {
-				ch <- cmdutil.UsageError(cmd, "There are no builds in the current project")
-				return
-			}
-			sort.Sort(sort.Reverse(api.ByCreationTimestamp(builds)))
-			msg := "A build name is required. Most recent builds:"
-			for i, b := range builds {
-				if i == maxBuildListLen {
-					break
-				}
-				msg += fmt.Sprintf("\n* %s\t%s\t%s ago", b.Name, b.Status.Phase, units.HumanDuration(time.Since(b.CreationTimestamp.Time)))
-			}
-			ch <- cmdutil.UsageError(cmd, msg)
-		}()
-		select {
-		case <-time.After(timeout):
-			return cmdutil.UsageError(cmd, "A build name is required")
-		case err := <-ch:
-			return err
-		}
+		return cmdutil.UsageError(cmd, "A build name is required - you can run `%s get builds` to list builds", fullName)
```

Comment:
the point is, if they ran the original command with a "-n" then we should include it on the sample command because clearly it's not the default for them (or they're specifying it unnecessarily).